src/blob_gc_job.cc

```diff
@@ -581,5 +581,73 @@ bool BlobGCJob::IsShutingDown() {
   return (shuting_down_ && shuting_down_->load(std::memory_order_acquire));
 }
 
+Status BlobGCJob::DigHole() {
+  Status s;
+  const auto& inputs = blob_gc_->sampled_inputs();
```

Comment:
Can we make sure that all the inputs from sampled_inputs() is to dig hole instead of compact? @NingLin-P

Reply:
actually we will have a gc_sampled_inputs for compact and a fs_sampled_inputs for free space

Reply:
Done.Add todo for sampled_inputs.